File: template_method/ruby/objective_purchase.rb
require './shopping_cart'
require './session.rb'

class ObjectivePurchase
    include ShoppingCart

    attr_accessor :session

    def initialize
      @session = Session.new
    end

    def register_user
      @purchase.user = @session.user_logged
    end

    def delivery(address)
      if address == nil
        @purchase.address = @session.user_logged.address
      else
        @purchase.address = address
      end
    end
end

File: template_method/en_us/java/ObjectivePurchase.java
/**
 * File: ObjectivePurchase
 * Role: Concrete Part
 * Purpose: In a objective purchase the user chooses not to register itself.
 */
public class ObjectivePurchase extends ShoppingCart {

	@Override
	public void registerUser() {
		// Do nothing
	}

	@Override
	public void setDeliveryAddress(){
		System.out.println("Delivery Address: 501 Auburn Ave NE, Atlanta, GA 30312, Estados Unidos");
	}

}
